### DISCUSSION TITLE
Why look to -z direction by default?

### DISCUSSION THREAD

**Discussion by llfab** [StereoKit/StereoKit#310] [Q&A]
By default positive x direction is right, positive y direction is up and depth is negative z direction. I wonder why that was chosen. I found some interesting aspects having this setup:
- Put a Window with a Pose.Identity and "walk" back a bit. You will see the window from the back
- Hence, a Identity-rotation will by default always show the "elements" facing away from you
- This also happens if you create a pose for a Window with a LookAtDir to you. It is facing away from you

I would be happy if you could explain the concept behind this and provide some "best practice" handling rules.

**Comment by maluoi** [comment]
The main reason for this coordinate system, is simply that it is the most common standard within this subset of XR! Both OpenXR and GLTF use a right handed coordinate system, but diverge a little on which direction along the Z axis is forward. OpenXR uses Z- as forward, and I felt like that was reasonable enough to stick with.

However, the math is set up to work correctly here. Identity rotation faces "Forward" (Z-), which, yes, is the same rotation as the user when they start the application. If you wish the user to face backwards on start, you could change the base rotation with `Renderer.CameraRoot`. If Z+ was forward, you'd still have the same situation with the user and the UI facing the same direction while at identity, just in the positive direction instead.

It sounds like you may also need to double check how you're using LookDir and LookAt, if you create a window pose using these functions, it will properly face the user. I do this pretty frequently :)

```csharp
Vec3 at = new Vec3(5,0,5);
Pose windowPose = new Pose( at, Quat.LookAt(from, Input.Head.position) );
```

**Comment by llfab** [comment]
Alright. Thanks. I also got the difference for the LookAt functions. In our CS forward is Z+ but Up is -Y to keep the right handed CS. As we have massive Geometry3D functions I tend to calculate everything in our "world" and just translate the resulting elements like to Vec3 etc. Still trying to figure out how that works best. Will keep you posted.
